components/infobox/extensions/wikis/starcraft2/infobox_extension_achievements.lua

```diff
@@ -0,0 +1,103 @@
+---
+-- @Liquipedia
+-- wiki=starcraft2
+-- page=Module:Infobox/Extension/Achievements
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local LeagueIcon = require('Module:LeagueIcon')
+local String = require('Module:StringUtils')
+local Team = require('Module:Team')
+
+local Opponent = require('Module:OpponentLibraries').Opponent
+
+local BASE_CONDITIONS = {
+	'[[liquipediatiertype::!Qualifier]]',
+	'[[liquipediatiertype::!Charity]]',
+	'[[liquipediatier::1]]',
+	'[[placement::1]]',
+	'[[prizepoolindex::1]]',
+}
+local GSL_ICON = 'GSLLogo_small.png'
+local GSL_CODE_A_ICON = 'GSL_CodeA.png'
+local CODE_A = 'Code A'
+
+local Achievements = {}
+
+function Achievements.team()
+	local pageName = mw.title.getCurrentTitle().text
+	local historicalPages = Team.queryHistoricalNames(pageName)
+	assert(historicalPages, 'No team template exists for "' .. pageName .. '"')
+	historicalPages = Array.extend(
+		Array.map(historicalPages, function(team) return (team:gsub(' ', '_')) end),
+		Array.map(historicalPages, function(team) return (team:gsub('_', ' ')) end)
+	)
+
+	return Achievements.display(Achievements._fetchDataForTeam(historicalPages, Opponent.team)),
+		Achievements.display(Achievements._fetchDataForTeam(historicalPages, Opponent.solo))
+end
+
+function Achievements._fetchDataForTeam(historicalPages, opponentType)
+	local lpdbKey = opponentType == Opponent.team and 'opponentname' or 'opponentplayers_p1team'
```

Comment:
This wouldn't find results where only one player of the team is playing together with some others, i assume that is by choice?

Reply:
yes
it is about solo achievements while on the team